watertap/unit_models/uv_aop.py

```diff
@@ -457,12 +504,39 @@ def eq_electricity_demand_phase_comp(b, t, p, j):
             )
 
         # TODO: add minimum electricity demand for multiple solutes
-        # @self.Constraint(
-        #     self.flowsheet().config.time,
-        #     doc="Constraints for minimum electricity demand of the UV reactor.",
-        # )
-        # def eq_minimum_electricity_demand(b, t):
-        #     return b.electricity_demand_minimum[t] == smooth_max(b.electricity_demand_phase_comp[t, p, j])
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.property_package.solute_set,
+            doc="Constraints for minimum electricity demand of the UV reactor.",
+        )
+        def eq_min_electricity_demand(b, t, j):
+            if j == b.config.property_package.solute_set.first():
+                return (
+                    b.electricity_demand_minimum[t, j]
+                    == b.electricity_demand_phase_comp[t, "Liq", j]
+                )
+            else:
+                return b.electricity_demand_minimum[t, j] == (
+                    smooth_max(
+                        b.electricity_demand_phase_comp[
+                            t, "Liq", b.config.property_package.solute_set.prev(j)
+                        ],
+                        b.electricity_demand_phase_comp[t, "Liq", j],
+                        b.eps_electricity,
+                    )
+                )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Constraints for total electricity demand of the UV reactor.",
+        )
+        def eq_electricity_demand(b, t):
+            return (
+                b.electricity_demand[t]
+                == b.electricity_demand_minimum[
+                    t, b.config.property_package.solute_set.last()
+                ]
+            )
```

Comment:
This whole set-up doesn't make a huge amount of sense to me from a mathematical modeling perspective. What quantity should `electricity_demand` be equal to?

If it's the `max(b.electricity_demand_phase_comp[t, "Liq", :])` then we need to be doing something different.

Reply:
All of these set-up is to avoid using `max` function because it is not continuous and may have problems when using IDAES ipopt solver. So I follow the way in this https://github.com/IDAES/idaes-pse/blob/d1a2add96bf660681048b05ea8e90fb4b51294ee/idaes/models/unit_models/mixer.py#L632

Reply:
This implementation is a bit different than the one you point to in IDAES, which looks correct to me.

I would also change the variable name (from min to max) to be less confusing.

I can think of another implementation which might be a bit more efficient, if you have a lot of values. @andrewlee94 I wonder if we should add a helper in IDAES to deal with these recursive min/max formulations, since we have at least two examples.

Reply:
We could add a helper function for this, although I am always a bit concerned about making things like this too easy to use for users who don't know the issues associated with them. PRs are always welcome.

On a different topic, I note that all the doc strings refer to a minimum energy demand whilst the function itself is a max. Whilst this does make some sense in the global process level if you stop and think about it for a while, in the more local sense you are really finding the maximum of the potential electricity demands. I think it might make more sense to call this `max_component_electricity_demand` or something similar to better indicate what this represents.

Reply:
Your second comment brings up a good point -- if this is really the "minimum" energy the process could use, but it could use more, then we could use a standard inequality representation and skip using `smooth_max`.

Reply:
@bknueven I am not sure that is a good idea - it leaves the energy unbounded to the upper end without any additional constraints. My initial though as I was reading this was that this would give us the minimum theoretical energy required, and the unit model could then add an efficiency onto this to get the actual energy demand - this would require that this be well-defined and not an inequality.

Reply:
I agree that we should change the name to `max_component_electricity_demand` because bas an intermedium vairable it is easier for understanding. @bknueven we talked about the inequality representation before, but we think it might have bounding issues when doing optimization problems.